eng/common/scripts/stress-testing/stress-test-deployment-lib.ps1

```diff
@@ -164,59 +176,86 @@ function DeployStressPackage(
-        Write-Host "  docker run -it $imageTag" -ForegroundColor DarkBlue
-        Write-Host "  docker run -it $imageTag <shell, e.g. 'bash' 'pwsh' 'sh'>" -ForegroundColor DarkBlue
-        Write-Host "To show installed container images:" -ForegroundColor Blue
-        Write-Host "  docker image ls" -ForegroundColor DarkBlue
-        Write-Host "To show running containers:" -ForegroundColor Blue
-        Write-Host "  docker ps" -ForegroundColor DarkBlue
-
-        Run docker push $imageTag
-        if ($LASTEXITCODE) {
-            if ($login) {
-                Write-Warning "If docker push is failing due to authentication issues, try calling this script with '-Login'"
+        $dockerFilePath = [System.IO.Path]::GetFullPath($dockerFilePath)
+        $dockerFilePaths += $dockerFilePath
+    }
+    
+
+    foreach ($dockerFilePath in $dockerFilePaths) {
+        if (!(Test-Path $dockerFilePath)) {continue}
+        $dockerfileName = ($dockerfilePath -split { $_ -in '\', '/' })[-1].ToLower()
+        $imageTag = $imageTagBase + "/${dockerfileName}:${deployId}"
+        if ($pushImages) {
+            Write-Host "Building and pushing stress test docker image '$imageTag'"
+            $dockerFile = Get-ChildItem $dockerFilePath
+            $dockerBuildFolder = if ($pkg.DockerBuildDir) {
+                Join-Path $pkg.Directory $pkg.DockerBuildDir
+            } else {
+                $dockerFile.DirectoryName
+            }
+            $dockerBuildFolder = [System.IO.Path]::GetFullPath($dockerBuildFolder).Trim()
+
+            Run docker build -t $imageTag -f $dockerFile $dockerBuildFolder
+            if ($LASTEXITCODE) { return }
+
+            Write-Host "`nContainer image '$imageTag' successfully built. To run commands on the container locally:" -ForegroundColor Blue
+            Write-Host "  docker run -it $imageTag" -ForegroundColor DarkBlue
+            Write-Host "  docker run -it $imageTag <shell, e.g. 'bash' 'pwsh' 'sh'>" -ForegroundColor DarkBlue
+            Write-Host "To show installed container images:" -ForegroundColor Blue
+            Write-Host "  docker image ls" -ForegroundColor DarkBlue
+            Write-Host "To show running containers:" -ForegroundColor Blue
+            Write-Host "  docker ps" -ForegroundColor DarkBlue
+
+            Run docker push $imageTag
+            if ($LASTEXITCODE) {
+                if ($login) {
+                    Write-Warning "If docker push is failing due to authentication issues, try calling this script with '-Login'"
+                }
+                return
             }
-            return
         }
+        ($genVal.scenarios | Where-Object {
+                $dockerPath = Join-Path $pkg.Directory $_.image;
+                [System.IO.Path]::GetFullPath($dockerPath) -eq
+                $dockerFilePath }).ForEach({$_.imageTag = $imageTag})
+        $genVal | ConvertTo-Yaml | Out-File -FilePath $genValFile
```

Comment:
Ordered

Reply:
ConvertTo-Yaml does not have ordered as parameter

Reply:
@ckairen Did you figure out the issue with randomized matrix outputs? When you were doing the demo it looked like the output order was getting messed up from what I would expect based on the input.

Reply:
Yup, the issue is with this [line](https://github.com/Azure/azure-sdk-tools/pull/3790/files#diff-c7f07b08bcb26f5a4d6d93abe4e64c5ec2b2cf7ad43a49e2326fc905b168e6ffR28) because convertFrom-Json doesn't have ordered hashtable. It's only in powershell 7.3, which is in preview [here's the link](https://github.com/MicrosoftDocs/PowerShell-Docs/pull/9042)